jerry-core/ecma/base/ecma-globals.h

```diff
@@ -683,13 +682,24 @@ typedef struct
     } array;
 
     /*
-     * Description of arguments objects.
+     * Description of pseudo array objects.
      */
     struct
     {
-      ecma_value_t lex_env_cp; /**< lexical environment */
-      uint32_t length; /**< length of names */
-    } arguments;
+      uint8_t type; /**< pseudo array type, e.g. Arguments, TypedArray*/
+      uint8_t extra_info; /**< extra infomations of the object.
+                            *  for TypedArray, it is element_width_shift */
```

Comment:
comment:
extra infomation about the object,
e.g. element_width_shift for typed arrays